### PR TITLE
[MNG-7568] Consider inactive profiles for validation

### PR DESCRIPTION
**Pull request by pzygielo**
I propose this cosmetic (key change is in new line 485, two other changes - CS:LineLength).
This allows to deactivate profiles from settings and from pom with the same id, and avoid confusing warning:
```
$ mvn -P \!XYZ verify
...
[WARNING] The requested profile "XYZ" could not be activated because it does not exist.
...
```

Similar issues
- MNG-7211 (done in unspecified version, while MNG-7051 fixed in v4)

--- 

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] SUMMARY`, where you replace `MNG-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by MartinKanters** [comment]
I think it's a nice change, but I'm not sure if it's fully compatible with the changes we did in master (for Maven 4). Can you verify what the behaviour is for Maven 4? Perhaps we have already solved this in a similar way.

File: maven-core/src/main/java/org/apache/maven/DefaultMaven.java [hunk 1]
```diff
@@ -466,7 +468,9 @@ private void validatePrerequisitesForNonMavenPluginProjects( List<MavenProject>
         }
     }
 
-    private void validateActivatedProfiles( List<MavenProject> projects, List<String> activeProfileIds )
+    private void validateActivatedProfiles( List<MavenProject> projects,
+                                            List<String> activeProfileIds,
+                                            List<String> inactiveProfiles )
```
Comment (MartinKanters) [inline maven-core/src/main/java/org/apache/maven/DefaultMaven.java:473]: Let's name this inactiveProfileIds for consistency

**Review by MartinKanters — CHANGES_REQUESTED** [review]
I think it's a nice change, but I'm not sure if it's fully compatible with the changes we did in master (for Maven 4). Can you verify what the behaviour is for Maven 4? Perhaps we have already solved this in a similar way.

File: maven-core/src/main/java/org/apache/maven/DefaultMaven.java (hunk 1 above)
Comment (pzygielo) [reply maven-core/src/main/java/org/apache/maven/DefaultMaven.java:473]: I followed the name of method used in the call. I agree that local naming style should be preserved here - will update that.

**Comment by pzygielo** [comment]
> Can you verify what the behaviour is for Maven 4? 

Using maven 4 (2a9f39336cec1d8e52d30cc48503d51ed8672536) it works fine - build succeeds and does not complain about missing profile. 

> Perhaps we have already solved this in a similar way.

I think there is no similar change needed there.

File: maven-core/src/main/java/org/apache/maven/DefaultMaven.java (hunk 1 above)
Comment (pzygielo) [reply maven-core/src/main/java/org/apache/maven/DefaultMaven.java]: This has been implemented.

**Comment by pzygielo** [comment]
> I think we should create a very small unit/IT for the sake of regression testing.

Here it is:
- https://github.com/apache/maven-integration-testing/pull/205

**Comment by michael-o** [comment]
This does not appear in master or 3.9.x?

**Comment by pzygielo** [comment]
> This does not appear in master or 3.9.x?

if (this == this issue) - no such problem in master.
Never heard of maven 3.9.x.

if (this == this change) - not proposed for master as there is no such problem.
I didn't test with maven 3.9.x. Feel free to pick this there is there is need.

**Comment by pzygielo** [comment]
Here is similar 
- https://github.com/apache/maven/pull/850

for 3.9.x.

**Review by gnodet — APPROVED** [review]

**Review by MartinKanters — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #8884

**Pull request merged**